### PR TITLE
[Merged by Bors] - feat: port NumberTheory.LegendreSymbol.Basic

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/LegendreSymbol/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+/-- If `legendreSym p a = -1`, then the only solution of `x^2 - a*y^2 = 0` in `ℤ/pℤ`
+is the trivial one. -/
+theorem eq_zero_mod_of_eq_neg_one {p : ℕ} [Fact p.Prime] {a : ℤ} (h : legendreSym p a = -1)
+    {x y : ZMod p} (hxy : x ^ 2 - a * y ^ 2 = 0) : x = 0 ∧ y = 0 := by
+  have ha : (a : ZMod p) ≠ 0 := by
+    intro hf
+    rw [(eq_zero_iff p a).mpr hf] at h
+    -- porting note `zero_ne_neg_of_ne` not ported? but no noalign? fixed by `tauto`
+    tauto
```
Comment (semorrison) [inline Mathlib/NumberTheory/LegendreSymbol/Basic.lean:257]: I agree we can probably skip porting this one: it was an internal implementation detail, and not that great as a public facing lemma.

`linarith` might be clearer here than `tauto`.

File: Mathlib/NumberTheory/LegendreSymbol/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+/-- The Legendre symbol `legendreSym p a = 1` if there is a solution in `ℤ/pℤ`
+of the equation `x^2 - a*y^2 = 0` with `x ≠ 0`. -/
+theorem eq_one_of_sq_sub_mul_sq_eq_zero' {p : ℕ} [Fact p.Prime] {a : ℤ} (ha : (a : ZMod p) ≠ 0)
+    {x y : ZMod p} (hx : x ≠ 0) (hxy : x ^ 2 - a * y ^ 2 = 0) : legendreSym p a = 1 := by
+  haveI hy : y ≠ 0 := by
+    rintro rfl
+    rw [zero_pow' 2 (by norm_num), MulZeroClass.mul_zero, sub_zero, pow_eq_zero_iff
+        (by norm_num : 0 < 2)] at hxy
+  -- porting note: apparently something was not automatically inferred and now is.
```
Comment (semorrison) [inline Mathlib/NumberTheory/LegendreSymbol/Basic.lean:244]:
```suggestion
```
I think it's fine to drop this. It was an unexpected failure (even with a comment asking about it) that is now just working. Hooray!

**Comment by semorrison** [comment]
bors d+

File: Mathlib/NumberTheory/LegendreSymbol/Basic.lean (hunk 1 above)
Comment (mo271) [reply Mathlib/NumberTheory/LegendreSymbol/Basic.lean]: done

**Pull request closed without merging**